src/plugins/data_source_management/public/components/direct_query_data_sources_components/associated_object_management/utils/associated_objects_tab_utils.tsx

```diff
@@ -43,49 +43,38 @@ export const isCatalogCacheFetching = (...statuses: DirectQueryLoadingStatus[])
   );
 };
 
-export const redirectToExplorerWithDataSrc = (
+export const redirectToDiscoverWithDataSrc = (
   datasourceName: string,
-  datasourceType: string,
+  datasourceMDSId: string | undefined,
   databaseName: string,
   tableName: string,
   application: ApplicationStart
 ) => {
-  const queryIndex = `${datasourceName}.${databaseName}.${tableName}`;
-  redirectToExplorerWithQuery(datasourceName, datasourceType, queryIndex, application);
-};
-
-export const redirectToExplorerOSIdx = (indexName: string, application: ApplicationStart) => {
-  redirectToExplorerWithQuery(
-    DEFAULT_DATA_SOURCE_NAME,
-    DEFAULT_DATA_SOURCE_TYPE,
-    indexName,
-    application
-  );
-};
-
-export const redirectToExplorerS3 = (datasourceName: string, application: ApplicationStart) => {
-  application.navigateToApp(observabilityLogsID, {
-    path: `#/explorer`,
-    state: {
-      datasourceName,
-      datasourceType: DATA_SOURCE_TYPES.S3Glue,
-    },
+  application.navigateToApp('data-explorer', {
+    path: `discover#?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_q=(filters:!(),query:(dataset:(dataSource:(id:'${
+      datasourceMDSId ?? ''
+    }',meta:(name:${datasourceName},type:CUSTOM),title:'',type:DATA_SOURCE),id:'${
+      datasourceMDSId ?? ''
+    }::${datasourceName}.${databaseName}.${tableName}',title:${datasourceName}.${databaseName}.${tableName},type:S3),language:SQL,query:'SELECT%20*%20FROM%20${datasourceName}.${databaseName}.${tableName}%20LIMIT%2010'))`,
   });
 };
 
-const redirectToExplorerWithQuery = (
-  datasourceName: string,
-  datasourceType: string,
-  queriedIndex: string,
+export const redirectToDiscoverOSIdx = (
```

Comment:
What does `redirectToDiscoverOSIdx` mean?

Reply:
redirect to discover with opensearch index. the naming scheme was exactly the same except i just replaced 'explorer' with 'discover'